src/Workspaces/Core/Portable/Workspace/Solution/ProjectDependencyGraph_RemoveProject.cs

```diff
@@ -42,74 +42,103 @@ internal ProjectDependencyGraph WithProjectRemoved(ProjectId projectId)
         /// <summary>
         /// Computes a new <see cref="_referencesMap"/> for the removal of a project.
         /// </summary>
-        /// <param name="referencesMap">The <see cref="_referencesMap"/> prior to the removal.</param>
+        /// <param name="existingForwardReferencesMap">The <see cref="_referencesMap"/> prior to the removal.</param>
         /// <param name="existingReverseReferencesMap">The <see cref="_lazyReverseReferencesMap"/> prior to the removal.
         /// This map serves as a hint to the removal process; i.e. it is assumed correct if it contains data, but may be
         /// omitted without impacting correctness.</param>
-        /// <param name="projectId">The ID of the project which is being removed.</param>
+        /// <param name="removedProjectId">The ID of the project which is being removed.</param>
         /// <returns>The <see cref="_referencesMap"/> for the project dependency graph once the project is removed.</returns>
         private ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> ComputeNewReferencesMapForRemovedProject(
-            ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> referencesMap,
+            ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> existingForwardReferencesMap,
             ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>>? existingReverseReferencesMap,
-            ProjectId projectId)
+            ProjectId removedProjectId)
         {
-            var builder = referencesMap.ToBuilder();
+            var builder = existingForwardReferencesMap.ToBuilder();
 
-            if (existingReverseReferencesMap is object && existingReverseReferencesMap.TryGetValue(projectId, out var referencingProjects))
+            if (existingReverseReferencesMap is object)
             {
                 // We know all the projects directly referencing 'projectId', so remove 'projectId' from the set of
                 // references in each of those cases directly.
-                foreach (var id in referencingProjects)
+                if (existingReverseReferencesMap.TryGetValue(removedProjectId, out var referencingProjects))
                 {
-                    builder[id] = builder[id].Remove(projectId);
+                    foreach (var id in referencingProjects)
+                    {
+                        var forwardReferences = builder[id].Remove(removedProjectId);
```

Comment:
Extract this and the "else" block into some common code?

Reply:
➡️ Included in #41288